app/pages/resource/reservation-calendar/time-slots/TimeSlots.js

```diff
@@ -24,30 +27,85 @@ class TimeSlots extends Component {
     time: PropTypes.string,
   };
 
+  calculatePlaceholders(selectedDate, slots) {
+    const firstTimeSlots = slots
+      .map(timeslots => timeslots && timeslots[0])
+      .filter(value => !!value && value.end);
+
+    if (firstTimeSlots.length === 0) {
+      return {
+        mobilePlaceholderOffset: 0,
+        timeSlotPlaceholderSizes: Array(slots.length).fill(0),
+      };
+    }
+
+    const earliestTimeSlot = minBy(firstTimeSlots, timeSlot =>
+      moment(timeSlot.start).format('HHMM')
+    );
+    const dateForTimeComparison = { year: 2000, dayOfYear: 1 };
+    const earliestStart = moment(earliestTimeSlot.start).set(dateForTimeComparison);
+
+    const timeSlotPlaceholderSizes = slots.map((slot) => {
+      if (!slot[0] || !slot[0].end) {
+        return null;
+      }
+      const currentStart = moment(slot[0].start).set(dateForTimeComparison);
+      return currentStart.diff(earliestStart, 'minutes') / 30;
+    });
+
+    const selectedDateIndex = findIndex(
+      slots,
+      slot => moment(slot[0].start).format(constants.DATE_FORMAT) === selectedDate
```

Comment:
When opening a resource from search result list, an error page flashes quickly and then it continues to the resource page normally. It happens because of this row's `slot[0].start` since slots array is not fully populated on first render. It needs an additional check or maybe using `firstTimeSlots` instead since it has prechecked values.

```suggestion
      slot => moment(slot.start).format(constants.DATE_FORMAT) === selectedDate
```

Reply:
Will check